include/transport/transport.h

```diff
@@ -3,118 +3,246 @@
 #include <mutex>
 #include <optional>
 #include <queue>
+#include <sys/socket.h>
 #include <vector>
 
 namespace qtransport {
 
-using TransportContextId = uint64_t;
-using MediaStreamId = uint64_t;
+using TransportContextId =
+    uint64_t; ///< Context Id is a 64bit number that is used as a key to maps
+using MediaStreamId = uint64_t; ///< Media stream Id is a 64bit number that is
+                                ///< used as a key to maps
 
-enum class TransportStatus : uint8_t { Ready = 0, Disconnected };
+/**
+ * Transport status/state values
+ */
+enum class TransportStatus : uint8_t {
+  Ready = 0,
+  Connecting,
+  RemoteRequestClose,
+  Disconnected
+};
 
```

Comment:
I'm not sure these states make sense to describe a server that at the same time lilstening for new connection, fully connected to some client, and in the process of connecting to another, and some other client has already disconnected